test/test_utils.py

```diff
@@ -806,6 +806,7 @@ def get_amp_supported_dtype():
 
 
 class TestExtensionUtils(TestCase):
+    @unittest.skip("TODO: somehow unregister xpu")
```

Comment:
Looking at the test logic, instead of skipping it, I think you could register the DummyXPUModule on line 822 with a different name, call it `lol` for example.  The test logic would be the same while it doesn't pull it a module called `xpu`

Reply:
There's a check that its a valid device type, instead I've sort of cleaned up the registration